File: sdk/src/main/java/org/maibot/sdk/manager/InteractionGroupManager.java
package org.maibot.sdk.manager;

import jakarta.persistence.EntityManager;
import org.maibot.sdk.storage.db.dao.InteractionGroup;

public interface InteractionGroupManager {
    InteractionGroup getOrCreatIfAbsent(
      EntityManager em,
      String platform,
      String platformGroupId,
      String groupName
    );

    InteractionGroup get(EntityManager em, String platform, String platformGroupId);
}

File: sdk/src/main/java/org/maibot/sdk/manager/InteractionGroupManager.kt
package org.maibot.sdk.manager

import jakarta.persistence.EntityManager
import org.maibot.sdk.storage.db.dao.InteractionGroup

interface InteractionGroupManager {
    fun getOrCreatIfAbsent(
        em: EntityManager,
        platform: String,
        platformGroupId: String,
        groupName: String?
    ): InteractionGroup?

    fun get(em: EntityManager, platform: String, platformGroupId: String): InteractionGroup?
}
